### PR TITLE
Replace TransferLearningScheduler with TransferLearningMixin

### PR DESCRIPTION
**Pull request by wistuba**
Replaces the TransferLearningScheduler in favor of a mixin. This will simplify implementing different types of transfer learning, e.g. transfer learning as part of a Hyperband RungSystem.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by geoalgo — COMMENTED** [review]
Looks good to me, have small comments.

File: syne_tune/optimizer/schedulers/transfer_learning/__init__.py [hunk 1]
```diff
@@ -49,17 +48,17 @@ def __init__(
...
         :param transfer_learning_evaluations: dictionary from task name to offline evaluations.
         :param metric_names: name of the metric to be optimized.
         """
-        super(TransferLearningScheduler, self).__init__(config_space=config_space)
         self._check_consistency(
             config_space=config_space,
             transfer_learning_evaluations=transfer_learning_evaluations,
             metric_names=metric_names,
         )
+        self._transfer_learning_evaluations = transfer_learning_evaluations
```
[the remark this answers is not in the document]
Comment (wistuba) [reply syne_tune/optimizer/schedulers/transfer_learning/__init__.py]: Is this what you had in mind?

[the remark this answers is not in the document]
Comment (geoalgo) [reply syne_tune/optimizer/schedulers/transfer_learning/__init__.py]: Yes exactly thanks.

File: syne_tune/optimizer/schedulers/transfer_learning/__init__.py [hunk 2]
```diff
@@ -79,3 +77,32 @@ def _check_consistency(
...
+        :param transfer_learning_evaluations: Set of candidates to choose from.
+        :param num_hyperparameters_per_task: The number of top hyperparameters per task to return.
+        :param mode: 'min' or 'max', indicating the type of optimization problem.
+        :param metric: The metric to consider for ranking hyperparameters.
+        :returns: Dict which maps from task name to list of hyperparameters in order.
+        """
+        assert num_hyperparameters_per_task > 0 and isinstance(num_hyperparameters_per_task, int), \
+            f"{num_hyperparameters_per_task} is no positive integer."
+        assert mode in ['min', 'max'], f"Unknown mode {mode}, must be 'min' or 'max'."
+        assert metric in self.metric_names(), f"Unknown metric {metric}."
```
Comment (geoalgo) [inline syne_tune/optimizer/schedulers/transfer_learning/__init__.py:98]: This also works, good idea.

**Review by geoalgo — APPROVED** [review]

**Pull request merged**